### PR TITLE
Create Google Static Map downloading service

### PR DESCRIPTION
**Pull request by marcelvanworkum**
Simple service which will be utilised by the GUI to display the rockets location on a map

### CODE REVIEW

File: software/map_downloader.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        self.session = requests.Session()
+
+        # read the api key from the os' exports
+        self.api_key = os.environ[self.API_ID]
+
+    def download_map(self):
+        # construct the request url
+        request = self.API_ROOT + self.center + self.API_SEPARATOR + self.ZOOM + \
+                  self.API_SEPARATOR + self.SIZE + self.API_SEPARATOR + \
+                  self.MAP_TYPE + self.API_SEPARATOR + self.marker + \
```
Comment (jamiesanson) [inline software/map_downloader.py]: Do these separators need to be constants? might be more concise without

File: software/map_downloader.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    API_SEPARATOR = "&"
+    SIZE = "size=400x400"
+    ZOOM = "zoom=16"
+    MAP_TYPE = "hybrid"
+    MARKER_TEMPLATE = "markers=color:red%7Clabel:R%7C"
+
+    def __init__(self, location, filename):
+        # initialize the location and center of map, setting a mark at the location
+        self.location = location
+        self.center = "center=" + location
```
Comment (jamiesanson) [inline software/map_downloader.py:36]: Should document the format of this location variable. Reading through it should me "lat,long", but should be commented somewhere

File: software/map_downloader.py (hunk 1 above)
Comment (marcelvanworkum) [reply software/map_downloader.py]: We I was thinking that, I guess the use of `&` as an API separator is probably prevalent enough to not be considered a magic string

**Comment by marcelvanworkum** [comment]
@jamiesanson made those code review changes, anything else or good to merge?

**Comment by jamiesanson** [comment]
Looks good to me. Fix le conflicts then merge when ready

**Pull request merged**